src/plugins/obstacle_avoidance_server/obstacle_avoidance_server_impl.cpp

```diff
@@ -0,0 +1,108 @@
+#include "obstacle_avoidance_server_impl.h"
+
+namespace mavsdk {
+
+ObstacleAvoidanceServerImpl::ObstacleAvoidanceServerImpl(System& system) : PluginImplBase(system)
+{
+    _parent->register_plugin(this);
+}
+
+ObstacleAvoidanceServerImpl::ObstacleAvoidanceServerImpl(std::shared_ptr<System> system) :
+    PluginImplBase(system)
+{
+    _parent->register_plugin(this);
+}
+
+ObstacleAvoidanceServerImpl::~ObstacleAvoidanceServerImpl()
+{
+    _parent->unregister_plugin(this);
+}
+
+void ObstacleAvoidanceServerImpl::init()
+{
+    using namespace std::placeholders;
+
+    _parent->register_mavlink_command_handler(
+        MAV_CMD_COMPONENT_CONTROL,
+        std::bind(&ObstacleAvoidanceServerImpl::process_component_control_command, this, _1),
+        this);
```

Comment:
We're trying to go away from `std::bind` because that's generally the recommendation, so I suggest:

```suggestion
    _parent->register_mavlink_command_handler(
        MAV_CMD_COMPONENT_CONTROL,
        [this](const MavlinkCommandReceiver::CommandLong& command) {
            process_component_control_command(command);
        });
```

Reply:
got it